sdk/python/kfp/_client.py

```diff
@@ -740,8 +761,9 @@ def run_pipeline(
 
     def create_recurring_run(
         self,
-        experiment_id: str,
         job_name: str,
+        experiment_id: Optional[str] = None,
```

Comment:
Moving the existing parameter could break users who call this method using positional arguments.

Reply:
That's a good point, I guess to satisfy the `python` constraint `args` must be before `kwargs` and make `experiment_id` optional I'd need to make `job_name` also a `kwarg`. I could validate that `job_name` is required if that'd work for you? Then we'd keep the same interface for existing users but allow for the case where either `experiment_id` or `experiment_name` is supplied.

Reply:
Reverted this change 😄